programs/ziti-edge-tunnel/instance-config.c

```diff
@@ -25,16 +25,14 @@
 #define MAX_BUFFER_LEN 1024 * 1024
 #define MIN_BUFFER_LEN 512
 
-static uv_mutex_t mutex;
-static bool mutex_initialized = false;
+static uv_sem_t sem;
+static unsigned int sem_value = 1;
+static int sem_initialized = -1;
 
 void initialize_instance_config() {
-    int mutex_init = uv_mutex_init(&mutex);
-    if (mutex_init != 0) {
-        ZITI_LOG(WARN, "Could not initialize lock for the config, config file may not be updated correctly");
-        mutex_initialized = false;
-    } else {
-        mutex_initialized = true;
+    int sem_initialized = uv_sem_init(&sem, sem_value);
```

Comment:
this variable is function-scope and shadows the module-scope declaration on line 30.

Reply:
fixed, thanks